pafy/backend_shared.py

```diff
@@ -401,6 +401,26 @@ def _sortkey(x, keybitrate=0, keyftype=0):
         else:
             return r
 
+    def getworstaudio(self, preftype="any", ftypestrict=True):
+        """ Return the lowest bitrate audio Stream object."""
+        if not self.audiostreams:
+            return None
+
+        def _sortkey(x, keybitrate=0, keyftype=0):
+            """ Sort function for min(). """
+            keybitrate = int(x.rawbitrate)
+            keyftype = preftype == x.extension
+            strict, nonstrict = (keyftype, keybitrate), (keybitrate, keyftype)
+            return strict if ftypestrict else nonstrict
```

Comment:
This nested function is exactly the same as defined in `getbestaudio` here:
https://github.com/mps-youtube/pafy/blob/933d08fc63d4ba1d82b7f10b3f8730ab9a664168/pafy/backend_shared.py#L384-L402

Perhaps we should move this sort key function to the current class scope instead of nesting the exact same definition inside two different methods. We could then use this sort key method from all other dependent methods without having to define it twice. This will help with code re-usability!